app/forms/user_form.rb

```diff
@@ -0,0 +1,26 @@
+class UserForm
+
+  include ActiveModel::Model
+
+  attr_accessor :first_name, :last_name, :password, :role, :email
+  validates :first_name, presence: true
+  validates :last_name, presence: true
+  validates :password, presence: true, length: { minimum: 8 }
+  validates :role, presence: true
+  validates :email, presence: true, length: { minimum: 8 }
+
+  def initialize(params= {})
+    @user = User.new
+    super(params)
+  end
+
+  def submit
+    return false if invalid?
+    @user.first_name = @first_name,
+    @user.last_name = @last_name,
+    @user.email = @email,
+    @user.role = @role,
+    @user.password = @password,

```

Comment:
I guess, it can be in a simpler way:
```ruby
@user.assign_attributes(attributes)
```
No need to update the PR. Just keep in mind that we in Ruby live shorter notations :)